lib/ast/function.rb

```diff
@@ -15,10 +15,19 @@ def code(entry = false)
       .concat finish_function(entry)
   end
 
+  def data
+    @clauses.map(&:data).join
+  end
+
   private
 
   def start_function(entry)
-    return "\n_#{name}:\n" if entry
+    if entry
+      return "\n" << <<~ASM
```

Comment:
there must be a nicer way to add a blank line a the start of a heredoc. or it could be add at the end of the previous bit

Reply:
for another time